ika/client.go

```diff
@@ -148,17 +155,17 @@ func (p *client) ApproveAndSign(
 		RequestType: "WaitForLocalExecution",
 	})
 	if err != nil {
-		return nil, fmt.Errorf("error executing transaction block: %w", err)
+		return nil, "", fmt.Errorf("error executing transaction block: %w", err)
 	}
-
+	txDigest := response.Effects.TransactionDigest
 	events, err := p.c.SuiGetEvents(ctx, models.SuiGetEventsRequest{
-		Digest: response.Effects.TransactionDigest,
+		Digest: txDigest,
 	})
 	if err != nil {
-		return nil, fmt.Errorf("error getting events: %w", err)
+		return nil, txDigest, fmt.Errorf("ika: %w", ErrEventParsing)
```

Comment:
let's do proper error wrapping, now we loose the underlying error (`err`).
Maybe we can use: https://pkg.go.dev/errors#Join
```
errors.Join(ErrEventParsing, err)
```

PS: you need to test if `Is` will handle it correctly (I think it does).

Reply:
It works correctly with `Is`. I updated the code